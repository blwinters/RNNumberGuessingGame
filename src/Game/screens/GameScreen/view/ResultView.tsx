import React from 'react'
import {Text} from 'react-native'
import {CheckGuessResult} from 'src/Game/types'

interface Props {
  result: CheckGuessResult | null
  guess: number
}

export default ({result, guess}: Props) => {
  if (result === 'tooLow') {
    return <Text>{guess} is too low.</Text>
  }

  if (result === 'tooHigh') {
    return <Text>{guess} is too high.</Text>
  }

  if (result === 'correct') {
    return <Text>That's correct! ðŸŽ‰</Text>
  }

  return <></>
}
