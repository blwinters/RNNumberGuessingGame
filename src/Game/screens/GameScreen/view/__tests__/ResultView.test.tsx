import React from 'react'
import {render, RenderAPI} from 'react-native-testing-library'
import ResultView from '../ResultView'

describe('ResetButton.tsx', () => {
  let instance: RenderAPI

  describe('when the result is `correct`', () => {
    beforeEach(() => {
      instance = render(<ResultView result={'correct'} guess={42} />)
    })
    it('should match snapshot', () => {
      expect(instance.toJSON()).toMatchSnapshot()
    })
  })

  describe('when the result is `tooLow`', () => {
    beforeEach(() => {
      instance = render(<ResultView result={'tooLow'} guess={42} />)
    })
    it('should match snapshot', () => {
      expect(instance.toJSON()).toMatchSnapshot()
    })
  })

  describe('when the result is `tooHigh`', () => {
    beforeEach(() => {
      instance = render(<ResultView result={'tooHigh'} guess={42} />)
    })
    it('should match snapshot', () => {
      expect(instance.toJSON()).toMatchSnapshot()
    })
  })
})
