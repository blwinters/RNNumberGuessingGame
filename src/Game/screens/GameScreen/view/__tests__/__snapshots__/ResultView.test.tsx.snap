// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetButton.tsx when the result is \`correct\` should match snapshot 1`] = `
<Text>
  That's correct! ðŸŽ‰
</Text>
`;

exports[`ResetButton.tsx when the result is \`tooHigh\` should match snapshot 1`] = `
<Text>
  42
   is too high.
</Text>
`;

exports[`ResetButton.tsx when the result is \`tooLow\` should match snapshot 1`] = `
<Text>
  42
   is too low.
</Text>
`;
